sw/device/tests/aes_force_prng_reseed_test.c

```diff
@@ -0,0 +1,509 @@
+   #         -Verify that the encryption stalls.
+   #         -Re-enable EDN. In order to do this it is required to first disable and re-enable the
+   #          CSRNG. In order to re-enable CSRNG, it is required to first disable and re-enable
+   #          the entropy source. The following sequence of steps should be used:
+   #             -Disable CSRNG
+   #             -Disable Entropy Source
+   #             -Enable Entropy Source
+   #             -Enable CSRNG
+   #             -Enable EDN
+   #         -Verify that the AES completes encryption.
+*/
+
+
+#define RESEED_TEST_EN 1  // Set to 1 to enable, 0 to disable
+#define DEBUG_EN 0  // Set to 1 to enable, 0 to disable
+#define TRYSETTING1_EN 1 // Set to 1 to enable, 0 to disable
+
+#include "sw/device/lib/dif/dif_aes.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/dif/dif_csrng.h"
+#include "sw/device/lib/dif/dif_entropy_src.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aes_testutils.h"
+#include "sw/device/lib/testing/csrng_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/sca/lib/simple_serial.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/multibits.h"
+#include "sw/device/lib/dif/dif_csrng_shared.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// Register Offsets based on the OpenTitan documentation
+#define AES_CTRL_SHADOWED_REG_OFFSET         0x74
+#define AES_CTRL_AUX_SHADOWED_REG_OFFSET     0x78
+#define AES_CTRL_AUX_REGWEN_REG_OFFSET       0x7C
+#define AES_TRIGGER_REG_OFFSET               0x80
+#define AES_STATUS_REG_OFFSET                0x84
+#define AES_CTRL_AUX_SHADOWED_KEY_TOUCH_FORCES_RESEED_BIT  0
+#define AES_CTRL_AUX_SHADOWED_FORCE_MASKS_BIT              1
+// Function to monitor and log the value of specific AES registers.
+static void monitor_aes_registers(mmio_region_t aes_base) {
+  uint32_t ctrl_shadowed = mmio_region_read32(aes_base, AES_CTRL_SHADOWED_REG_OFFSET);
+  uint32_t ctrl_aux_shadowed = mmio_region_read32(aes_base, AES_CTRL_AUX_SHADOWED_REG_OFFSET);
+  uint32_t ctrl_aux_regwen = mmio_region_read32(aes_base, AES_CTRL_AUX_REGWEN_REG_OFFSET);
+  uint32_t trigger = mmio_region_read32(aes_base, AES_TRIGGER_REG_OFFSET);
+  uint32_t status = mmio_region_read32(aes_base, AES_STATUS_REG_OFFSET);
+
+  LOG_INFO("REGISTER AES_CTRL_SHADOWED: 0x%x", ctrl_shadowed);
+  LOG_INFO("REGISTER AES_CTRL_AUX_SHADOWED: 0x%x", ctrl_aux_shadowed);
+  LOG_INFO("REGISTER AES_STATUS: 0x%x", status);
+  LOG_INFO("REGISTER AES_TRIGGER: 0x%x", trigger);
+  LOG_INFO("REGISTER AES_CTRL_AUX_REGWEN: 0x%x", ctrl_aux_regwen);
+}
+
+
+enum {
+  kTestTimeout = 1000 * 1000,  // Timeout for waiting for AES operations
+};
+
+static  dif_aes_t aes;
+#if RESEED_TEST_EN
```

Comment:
Please remove #defines. These are not allowed by the style guide. Please also remove all other #defines in this PR.

https://opentitan.org/book/doc/contributing/style_guides/c_cpp_coding_style.html